src/celldega/pre/__init__.py

```diff
@@ -1223,6 +1389,406 @@ def _check_required_files(technology, data_dir):
+    if matrix_root is None:
+        raise FileNotFoundError("matrix_files directory is missing")
+    slice_name = matrix_root.name.split("_")[0]
+    return dataset, slice_name
+
+
+def find_spot_positions(
+    data_dir: str, path_landscape_files: str, image_scale: float = 1.0
+) -> None:
+    """Generate IST cell centroids in pixel coordinates."""
+    dataset, inst_slice = _parse_ist_names(data_dir)
+    gc = pd.read_csv(
+        Path(data_dir) / "registered_images" / f"globalpos_{dataset}.csv", index_col=0
+    )
+    x_shift = gc.loc[f"{inst_slice}_{dataset}", "X_shift"]
+    y_shift = gc.loc[f"{inst_slice}_{dataset}", "Y_shift"]
+
+    # Load "raw" barcodes which give spot position
+    barcodes_path = (
+        Path(data_dir)
+        / "matrix_files"
+        / f"{inst_slice}_{dataset}"
+        / f"{inst_slice}_{dataset}_raw"
+        / "barcodes.tsv.gz"
+    )
+    barcodes = pd.read_csv(barcodes_path, sep="\t", header=None, index_col=0)
+    barcodes.index.name = None
+    barcodes["x"] = pd.NA
+    barcodes["y"] = pd.NA
+
+    map_file = Path(data_dir) / f"Substrate_{dataset}_map_file.tsv"
+    if map_file.exists():
+        chunk_size = 10_000_000
+        barcodes_set = set(barcodes.index.tolist())
+        for chunk in pd.read_csv(
+            map_file, sep="\t", chunksize=chunk_size, header=None, index_col=0
+        ):
+            common = list(barcodes_set.intersection(chunk.index.tolist()))
+            if common:
+                print('found common barcodes:', len(common))
+                barcodes.loc[common, "x"] = chunk.loc[common, 1]
+                barcodes.loc[common, "y"] = chunk.loc[common, 2]
+
+    scale = 1 / 0.382
+
+    # barcodes are in nm and need to be converted to microns
+    barcodes["x"] = (barcodes["x"].astype(float)/1000 - x_shift) * scale
+    barcodes["y"] = (barcodes["y"].astype(float)/1000 - y_shift) * scale
+
+    barcodes["name"] = barcodes.index.astype(str)
+    barcodes["geometry"] = barcodes.apply(
+        lambda r: [r["y"] / image_scale, r["x"] / image_scale], axis=1
+    )
+
+    print('spot positions to parquet')
+    barcodes[["name", "geometry"]].to_parquet(
+        Path(path_landscape_files) / "spot_positions.parquet"
+    )
+
+
+
+# Polars/Pandas version - testing
+def make_pseudo_transcript_tiles(
+    paths: dict[str, str],
+    path_spot_positions: str,
+    path_output: str,
+    tile_size: int,
+    jitter: float = 1.0,
+    *,
+    rng: np.random.Generator | None = None,
+) -> dict:
+    print('========Make pseudo transcript tiles========')
+
+    # Load spot positions
+    spots = pd.read_parquet(path_spot_positions)
+    spots[["y", "x"]] = spots["geometry"].apply(pd.Series)
+    spots = spots.set_index("name")[["x", "y"]]
+
+    print('range of x and y values in spots:')
+    print('x:', spots.x.min(), spots.x.max())
+    print('y:', spots.y.min(), spots.y.max())
+
+    # Read sparse cell-by-gene matrix
+    sbg = read_cbg_mtx(paths['sbg_matrix'], 'Xenium')
```

Comment:
@cornhundred should this be `IST` instead?

Reply:
This is because of the way that spot/cell names are encoded in IST data - this can be cleaned up